prov/rxd/src/rxd_cq.c

```diff
@@ -307,7 +307,18 @@ void rxd_progress_tx_list(struct rxd_ep *ep, struct rxd_peer *peer)
 			}
 			continue;
 		}
-
+				
+		if (tx_entry->op == RXD_DATA_READ && !tx_entry->bytes_done) {
+			if (ep->peers[tx_entry->peer].unacked_cnt >=
+		    	    ep->peers[tx_entry->peer].tx_window) {
+				break;
+			} else {
+				tx_entry->start_seq = ep->peers[tx_entry->peer].tx_seq_no;
+				ep->peers[tx_entry->peer].tx_seq_no = tx_entry->start_seq +
+							      	      tx_entry->num_segs;
+			}
```

Comment:
The else statement simply relocates the existing code to a new location.  But I don't understand why the new code was added.

Reply:
We need to make sure that this is only done if we end up in the while loop which is protected by the check that is being added.

Reply:
Yeah, I still don't understand, and it's not clear from the commit message.